clippy_lints/src/octal_escapes.rs

```diff
@@ -0,0 +1,131 @@
+use clippy_utils::diagnostics::span_lint_and_then;
+use rustc_ast::ast::{Expr, ExprKind};
+use rustc_ast::token::{Lit, LitKind};
+use rustc_errors::Applicability;
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_middle::lint::in_external_macro;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::Span;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for `\0` escapes in string and byte literals that look like octal character
+    /// escapes in C
```

Comment:
```suggestion
    /// escapes in C.
```
(Really NIT picky :sweat_smile: )

Reply:
That's what clippy is all about :)